# 데이터 통합

In [1]:
# Jupyter Notebook 화면 크기 조정 (width 값)

from IPython.core.display import display, HTML
display(HTML("<style>.container { width:80% !important; }</style>"))

## 1. Import Libraries

In [2]:
import time

import pandas as pd
import numpy as np

## 2. Import csv files

In [3]:
cine_df = pd.read_csv("./cine21_final_df.csv")
cine_df

title      release_date    genre  \
0                         0.0MHz  개봉일 : 2019-05-29       공포   
1                       10 미니츠 곤     개봉일 : 2019-11   범죄, 액션   
2     100일 동안 100가지로 100퍼센트 행복찾기  개봉일 : 2019-09-12      코미디   
3                        12번째 솔저  개봉일 : 2019-04-11       전쟁   
4                        13년의 공백  개봉일 : 2019-07-04  드라마, 가족   
...                          ...               ...      ...   
1806                     호라이즌 라인  개봉일 : 2021-02-10      스릴러   
1807                   호스트: 접속금지  개봉일 : 2021-07-21       공포   
1808                   호스트: 접속금지  개봉일 : 2021-07-21       공포   
1809                   혼자 사는 사람들  개봉일 : 2021-05-19      드라마   
1810                   혼자 사는 사람들  개봉일 : 2021-05-19      드라마   

                    rating                                              story  \
0     Unable to get rating  에디슨의 마지막 발명품, 유령 탐지기?“증명할 수 없는 미스터리란 없다”가지 말라는...   
1     Unable to get rating  룰이 깨진 순간, 모든 멤버가 타겟이 된다!동물적인 직감과 1초의 오차도 용납하지 ...   
2     Unable to get rating  어렸을 적부터 함께 자라고, IT 회사도 공동 운영,사는 집도 아랫집, 윗집으로 이...   
3                 12세이상관람가  제2차 세계대전 나치에 점령된 노르웨이.나치를 함락시킬 유일한 ‘마틴 레드 작전’에...   
4                 12세이상관람가  “아버지가 너무 싫습니다.그런데, 조금은 좋아하는 것 같기도 합니다.”담배를 사러 ...   
...                    ...                                                ...   
1806              12세이상관람가  계기판도 GPS도 조종사도 없다!내 안의 생존 본능을 깨워라!에메랄드 빛 해안을 낀...   
1807              12세이상관람가  전 세계가 극찬한 100% 리얼 팬데믹 호러!지금, 당신의 랜선미팅에 무언가가 접속...   
1808              12세이상관람가  전 세계가 극찬한 100% 리얼 팬데믹 호러!지금, 당신의 랜선미팅에 무언가가 접속...   
1809              12세이상관람가  집에서도 밖에서도 늘 혼자가 편한 진아.사람들은 자꾸 말을 걸어오지만, 진아는 그저...   
1810              12세이상관람가  집에서도 밖에서도 늘 혼자가 편한 진아.사람들은 자꾸 말을 걸어오지만, 진아는 그저...   

                                                   url  write_date score  \
0     http://www.cine21.com/movie/info/?movie_id=52988  2019-06-02     5   
1     http://www.cine21.com/movie/info/?movie_id=55539  2019-12-28     2   
2     http://www.cine21.com/movie/info/?movie_id=55031  2019-09-22     6   
3     http://www.cine21.com/movie/info/?movie_id=54607  2019-04-27     7   
4     http://www.cine21.com/movie/info/?movie_id=54778         NaN   NaN   
...                                                ...         ...   ...   
1806  http://www.cine21.com/movie/info/?movie_id=57226  2021-02-22     4   
1807  http://www.cine21.com/movie/info/?movie_id=57001  2021-08-01     6   
1808  http://www.cine21.com/movie/info/?movie_id=57001  2021-08-01     5   
1809  http://www.cine21.com/movie/info/?movie_id=57892  2021-05-23    10   
1810  http://www.cine21.com/movie/info/?movie_id=57892  2021-05-19     7   

                                                 review  
0          90년대 중반 심은하 주연 납량 특집드라마 'M' 을 다시 보는 느낌이 들었다.  
1     전개도 너무 시시한데다 결말도 엄청 이상하게 지어서 내용이 뭐가 어떻게 돌아가는지 ...  
2                        뻔한 재미와 감동이지만 시츄에이션 설정이 독창적이었다.  
3     제2차 세계대전 때, 나치로부터 노르웨이를 되찾기 위한 작전을 펼친 노르웨이 비밀 ...  
4                                                   NaN  
...                                                 ...  
1806                       재난 스릴러물은데 전혀 긴장감이 안 느껴지는게 문제  
1807            맨 마지막 장면을 위한 빌드업이 조금 루즈했지만 마지막은 진짜 놀랐다.  
1808  <호스트:접속금지>는 코로나 시대에 자가 격리 된 여섯 명의 친구가 화상채팅 프로그...  
1809  시대가 좋아졌다고는 하지만 좋아진 시대를 누리는 이들은 많지 않으며  좋든 싫든 인...  
1810                        혼자 살고 비슷한 성향이라 많이 공감이 가는 영화  

[1811 rows x 9 columns]

In [30]:
# 씨네21에서 리뷰가 4건 이상
tmp = cine_df.title.value_counts()
len(tmp[tmp > 4])

53

In [5]:
naver19 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/naver_review2019.csv")
naver20 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/naver_review2020.csv")
naver21 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/naver_review2021.csv")

In [6]:
print(len(naver19))
print(len(naver20))
print(len(naver21))
# 총 371429행

124672
185160
61597


In [14]:
naver20

<<<<<<< HEAD
title                    score content                                            write_date       release_date    source_site
100% 울프: 푸들이 될 순 없어      10    애초에 애들 보라고 만든 애니메이션인데 어른들 보기에 재미없다고 평점 깎는건 뭔 ..... 2020.12.27 03:10 2020 .12.24 개봉        naver
                               완전 재밌게 봤어요~ 저도 재밌었지만 초등인 저희 아이들은 진짜 신나게 봤네요~ ^^    2020.12.26 03:16 2020 .12.24 개봉        naver
                               어른도 재미지게 볼 수 있는 영화였어요. 아이도 박장대소 하더라구요. 연말 행복해 지... 2020.12.27 13:33 2020 .12.24 개봉        naver
                               보름달 밤 '늑대'로 변신하는 늑대인간들의 이야기에 가족의 사랑, 강아지들과의 우정이... 2020.12.24 10:54 2020 .12.24 개봉        naver
...                                                                                                                        ...
힐빌리의 노래                  6.0   미국에서도 개천에서 용난다는 얘기?? 나 이렇게 불우햇지만 열심히 공부했다?? 보는내... 2020.11.11 19:34 2020 .11.24 개봉        naver
                         2.0   배우들의 연기력이 아까울 정도로 평작. 넷플렉스의. 올해의 실수. 글렌클로즈와 에이미... 2020.11.28 23:09 2020 .11.24 개봉        naver
                               위대한 원작을 PC 로 쩔어버린 원작이 말하고자 하는 아메리칸 드림의 가치를 송두리채... 2021.04.27 00:47 2020 .11.24 개봉        naver
                         6.0   클라이막스 없이 오롯이 연기력과 답답하게 속터지는 감정만 남았고. 역시 한국영화가 최고다  2020.11.14 00:31 2020 .11.24 개봉        naver
>>>>>>> detail_get_movie NaN   NaN                                                NaN              NaN                     NaN

[185160 rows x 1 columns]

In [12]:
naver_df = naver19
naver_df = naver_df.append(naver20)
naver_df = naver_df.append(naver21)
naver_df

title  score                                            content  \
0      0.0MHz    1.0  와 정말 무서운집 수준의 공포영화가 한국에 다시 나왔네요. 꼭 보러 가세요.착석하신...   
1      0.0MHz   10.0     내용 구성이나 연출, cg는 저예산 영화 정도의 수준이었지만 정은지가 연기를 잘한다   
2      0.0MHz    2.0  상엽이 형은 어쩌다가 어릴때 죽었는지 다리 피흘리는 여고생 연관있을줄 알았는데 연관...   
3      0.0MHz   10.0                                       정은지 연기 소름...   
4      0.0MHz    6.0  일단 정은지는 완전 합격점. 최윤영도 좋았다. 여배우 둘 빼면 에휴...게다가 초반...   
...       ...    ...                                                ...   
61592   흩어진 밤   10.0                           스포일러가 포함된 감상평입니다. 감상평 보기   
61593   흩어진 밤   10.0                                    I love you 180!   
61594   흩어진 밤   10.0                 어느 누구 한명의 잘못도 아닌데 모두가 흩어져야 하는 밤...   
61595   흩어진 밤    1.0       진짜 연기력도 그렇고 제작능력이며 스토리며 다 없어요,,,,감동도 재미도 ,,,   
61596   흩어진 밤    2.0  전주국제영화제를 너무너무 싫어하고 신뢰하지 않는 영화제라고 생각한다. 여기서 대상을...   

             write_date    release_date source_site <<<<<<< HEAD  
0      2019.05.29 13:30  2019 .05.29 개봉       naver          NaN  
1      2019.05.29 10:58  2019 .05.29 개봉       naver          NaN  
2      2019.05.29 20:15  2019 .05.29 개봉       naver          NaN  
3      2019.05.29 12:21  2019 .05.29 개봉       naver          NaN  
4      2019.05.29 11:34  2019 .05.29 개봉       naver          NaN  
...                 ...             ...         ...          ...  
61592  2021.06.25 20:59  2021 .06.24 개봉       naver          NaN  
61593  2021.06.25 07:54  2021 .06.24 개봉       naver          NaN  
61594  2021.06.24 11:43  2021 .06.24 개봉       naver          NaN  
61595  2021.06.27 16:10  2021 .06.24 개봉       naver          NaN  
61596  2021.08.17 13:52  2021 .06.24 개봉       naver          NaN  

[371429 rows x 7 columns]

In [9]:
daum19 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/daum_review2019_last.csv")
daum20 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/daum_review2020_last.csv")
daum21 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/daum_review2021_last.csv")

In [47]:
print(len(daum19))
print(len(daum20))
print(len(daum21))
# 총 51773행

21896
14580
15297


In [46]:
daum_df = daum19
daum_df = daum_df.append(daum20)
daum_df = daum_df.append(daum21)
daum_df

title  score                                            content  \
0      모가디슈      9  개인적으로 오랜만에 잘 만들어진  한국영화를 본 기분이 들어서 좋았습니다. 가장 돋...   
1      모가디슈     10  어제 밤에 극장에서 보면서 다른 걸 떠나서 다시 한번 '이게 가능해?'라는 생각이 ...   
2      모가디슈     10  딱히 좋아하는 배우는 안나오지만 류감독님 영화라 믿고 봤어요보고나서 김윤석 배우님 ...   
3      모가디슈     10  오늘 조조로 보고 왔는데 영화 잘 만들었어오. 시종일관 긴장감을 잘 이끌어가고  마...   
4      모가디슈     10  오늘 아이맥스에서 봤습니다. 정말 마지막 장면에 와이프하고 같이 울컥했습니다.  1...   
...     ...    ...                                                ...   
15292    서복      1                                                NaN   
15293    서복      1                                                NaN   
15294    서복      6                                                NaN   
15295    서복      4                                                NaN   
15296    서복      1                                                NaN   

                write_date release_date source_site  Unnamed: 0  
0      2021. 07. 28. 15:20          NaN        daum         NaN  
1      2021. 07. 29. 10:43          NaN        daum         NaN  
2      2021. 07. 29. 11:30          NaN        daum         NaN  
3      2021. 07. 28. 21:15          NaN        daum         NaN  
4      2021. 07. 28. 22:33          NaN        daum         NaN  
...                    ...          ...         ...         ...  
15292  2021. 04. 18. 01:11          NaN        daum     15292.0  
15293  2021. 04. 17. 22:08          NaN        daum     15293.0  
15294  2021. 04. 19. 08:49          NaN        daum     15294.0  
15295  2021. 04. 17. 22:17          NaN        daum     15295.0  
15296  2021. 04. 18. 22:52          NaN        daum     15296.0  

[51773 rows x 7 columns]

In [10]:
watcha19 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/Wa_review2019_last.csv")
watcha20 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/Wa_review2020_last.csv")
watcha21 = pd.read_csv("C:/Users/Danbee/Desktop/sample-project/server/flask/utils/data/review/Wa_review2021_last.csv")

In [33]:
print(len(watcha19))
print(len(watcha20))
print(len(watcha21))
# 총 6825행

1159
4590
1076


In [45]:
watcha_df = watcha19
watcha_df = watcha_df.append(watcha20, ignore_index=True)
watcha_df = watcha_df.append(watcha21, ignore_index=True)
watcha_df

title score                                            content  \
0     나의 특별한 형제   2.0                  어린 어른의 무책임한 책임, 그럼에도 남아있는 희망의 불씨.   
1     나의 특별한 형제   2.5  신하균은 튀고 이솜은 의아한 와중에이광수가 떠받들고있는 모양새제작진도 주제 면에서어...   
2     나의 특별한 형제   2.0                                    장애인과 비장애인으로 나눈다   
3     나의 특별한 형제   3.0  디질래 맞짱뜨까 지예 바보 아니다.이거 보고 울 줄 알았냐. 이쒸.. 매너없게 쿠키...   
4     나의 특별한 형제   3.0                                 시리, 너 답답할 때 많았어...   
...         ...   ...                                                ...   
6820      미드나이트   2.5                                    112 문자신고 가능합니다.   
6821      미드나이트   0.5  배우들 열연 (이라고 나는 생각한다.) 을 제외하면 장점이 전혀없다.한국 영화계를 ...   
6822      미드나이트   1.5                                                NaN   
6823      미드나이트   2.5  문희: 호↗!!!박↗!!!고↗!!!구↗!!!마아아아아아↗!!! 이제 됐냐!! 이ㅆ....   
6824      미드나이트   2.0                                          딱 처음만 볼만함   

      write_date  source_site  
0            NaN  watchapedia  
1            NaN  watchapedia  
2            NaN  watchapedia  
3            NaN  watchapedia  
4            NaN  watchapedia  
...          ...          ...  
6820         NaN  watchapedia  
6821         NaN  watchapedia  
6822         NaN  watchapedia  
6823         NaN  watchapedia  
6824         NaN  watchapedia  

[6825 rows x 5 columns]

cine_df 1811 rows <br>
naver_df 371429 rows <br>
daum_df 51773 rows <br>
watcha 6825 rows <br>

all_df 431838 rows

In [54]:
naver_set = set(naver_df['title'])
len(naver_set)

1153

In [55]:
daum_set = set(daum_df['title'])
len(daum_set)

96

In [56]:
watcha_set = set(watcha_df['title'])
len(watcha_set)

411

In [52]:
cine_set = set(cine_df['title'])
len(cine_set)

1127

In [61]:
unique_titles = naver_set.intersection(daum_set, watcha_set, cine_set)
print(len(unique_titles))
unique_titles

29


{'007 노 타임 투 다이',
 '고질라 VS. 콩',
 '극장판 귀멸의 칼날: 무한열차편',
 '극장판 짱구는 못말려: 격돌! 낙서왕국과 얼추 네 명의 용사들',
 '극장판 포켓몬스터: 정글의 아이, 코코',
 '더 수어사이드 스쿼드',
 '더 스파이',
 '루카',
 '모가디슈',
 '미나리',
 '미드나이트',
 '미션 파서블',
 '발신제한',
 '방법: 재차의',
 '보스 베이비 2',
 '비와 당신의 이야기',
 '새해전야',
 '서복',
 '소울',
 '싱크홀',
 '올드',
 '이스케이프 룸 2: 노 웨이 아웃',
 '자산어보',
 '정글 크루즈',
 '캐시트럭',
 '컨저링 3: 악마가 시켰다',
 '크루즈 패밀리: 뉴 에이지',
 '토이 스토리 4',
 '프리 가이'}

In [59]:
print(naver_df.title.nunique())
print(daum_df.title.nunique())
print(watcha_df.title.nunique())
print(cine_df.title.nunique())

1152
96
411
1127


In [49]:
all_df = naver_df
all_df = all_df.append(daum_df, ignore_index=True)
all_df = all_df.append(watcha_df, ignore_index=True)
all_df = all_df.append(cine_df, ignore_index=True)
all_df

title score                                            content  \
0          0.0MHz     1  와 정말 무서운집 수준의 공포영화가 한국에 다시 나왔네요. 꼭 보러 가세요.착석하신...   
1          0.0MHz    10     내용 구성이나 연출, cg는 저예산 영화 정도의 수준이었지만 정은지가 연기를 잘한다   
2          0.0MHz     2  상엽이 형은 어쩌다가 어릴때 죽었는지 다리 피흘리는 여고생 연관있을줄 알았는데 연관...   
3          0.0MHz    10                                       정은지 연기 소름...   
4          0.0MHz     6  일단 정은지는 완전 합격점. 최윤영도 좋았다. 여배우 둘 빼면 에휴...게다가 초반...   
...           ...   ...                                                ...   
431833    호라이즌 라인     4                                                NaN   
431834  호스트: 접속금지     6                                                NaN   
431835  호스트: 접속금지     5                                                NaN   
431836  혼자 사는 사람들    10                                                NaN   
431837  혼자 사는 사람들     7                                                NaN   

              write_date      release_date source_site <<<<<<< HEAD  \
0       2019.05.29 13:30    2019 .05.29 개봉       naver          NaN   
1       2019.05.29 10:58    2019 .05.29 개봉       naver          NaN   
2       2019.05.29 20:15    2019 .05.29 개봉       naver          NaN   
3       2019.05.29 12:21    2019 .05.29 개봉       naver          NaN   
4       2019.05.29 11:34    2019 .05.29 개봉       naver          NaN   
...                  ...               ...         ...          ...   
431833        2021-02-22  개봉일 : 2021-02-10         NaN          NaN   
431834        2021-08-01  개봉일 : 2021-07-21         NaN          NaN   
431835        2021-08-01  개봉일 : 2021-07-21         NaN          NaN   
431836        2021-05-23  개봉일 : 2021-05-19         NaN          NaN   
431837        2021-05-19  개봉일 : 2021-05-19         NaN          NaN   

        Unnamed: 0 genre    rating  \
0              NaN   NaN       NaN   
1              NaN   NaN       NaN   
2              NaN   NaN       NaN   
3              NaN   NaN       NaN   
4              NaN   NaN       NaN   
...            ...   ...       ...   
431833         NaN   스릴러  12세이상관람가   
431834         NaN    공포  12세이상관람가   
431835         NaN    공포  12세이상관람가   
431836         NaN   드라마  12세이상관람가   
431837         NaN   드라마  12세이상관람가   

                                                    story  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
431833  계기판도 GPS도 조종사도 없다!내 안의 생존 본능을 깨워라!에메랄드 빛 해안을 낀...   
431834  전 세계가 극찬한 100% 리얼 팬데믹 호러!지금, 당신의 랜선미팅에 무언가가 접속...   
431835  전 세계가 극찬한 100% 리얼 팬데믹 호러!지금, 당신의 랜선미팅에 무언가가 접속...   
431836  집에서도 밖에서도 늘 혼자가 편한 진아.사람들은 자꾸 말을 걸어오지만, 진아는 그저...   
431837  집에서도 밖에서도 늘 혼자가 편한 진아.사람들은 자꾸 말을 걸어오지만, 진아는 그저...   

                                                     url  \
0                                                    NaN   
1                                                    NaN   
2                                                    NaN   
3                                                    NaN   
4                                                    NaN   
...                                                  ...   
431833  http://www.cine21.com/movie/info/?movie_id=57226   
431834  http://www.cine21.com/movie/info/?movie_id=57001   
431835  http://www.cine21.com/movie/info/?movie_id=57001   
431836  http://www.cine21.com/movie/info/?movie_id=57892   
431837  http://www.cine21.com/movie/info/?movie_id=57892   

                                                   review  
0                                                     NaN  
1                                                     NaN  
2                                                     NaN  
3                                                     NaN  
4          

In [68]:
all_df.sort_values('title', ignore_index=True)

title score                                            content  \
0       #살아있다     2                                                NaN   
1       #살아있다     1  넷xxx로 봤는데\r\n참 나 원\r\n욕하고 싶어서로긴한건 처음\r\n욕을 평점으...   
2       #살아있다     9  그리 길지않고  참 재밋게 넷에서\r\n 어젯밤 11시 30분 부터봤는데..무서웠음...   
3       #살아있다    10  뭔가 거창하지 하지않고 우리동네인데...몰입감 있게 잘 만들었네~ 생존일수를 좀 줄...   
4       #살아있다     5  아무리 좀비영화가 떳다 해도 좋은 배우들을 써먹어 이렇게 막만들다니... 능력없는 ...   
...       ...   ...                                                ...   
431833    NaN   NaN                                                NaN   
431834    NaN   NaN                                                NaN   
431835    NaN   NaN                                                NaN   
431836    NaN   NaN                                                NaN   
431837    NaN   NaN                                                NaN   

                 write_date release_date source_site <<<<<<< HEAD  Unnamed: 0  \
0       2020. 09. 12. 00:33          NaN        daum          NaN         NaN   
1       2020. 09. 13. 20:31          NaN        daum          NaN         NaN   
2       2020. 09. 13. 20:44          NaN        daum          NaN         NaN   
3       2020. 09. 13. 21:33          NaN        daum          NaN         NaN   
4       2020. 09. 13. 22:00          NaN        daum          NaN         NaN   
...                     ...          ...         ...          ...         ...   
431833                  NaN          NaN         NaN        naver         NaN   
431834                  NaN          NaN         NaN        naver         NaN   
431835                  NaN          NaN         NaN        naver         NaN   
431836                  NaN          NaN         NaN        naver         NaN   
431837                  NaN          NaN         NaN          NaN         NaN   

       genre rating story  url review  
0        NaN    NaN   NaN  NaN    NaN  
1        NaN    NaN   NaN  NaN    NaN  
2        NaN    NaN   NaN  NaN    NaN  
3        NaN    NaN   NaN  NaN    NaN  
4        NaN    NaN   NaN  NaN    NaN  
...      ...    ...   ...  ...    ...  
431833   NaN    NaN   NaN  NaN    NaN  
431834   NaN    NaN   NaN  NaN    NaN  
431835   NaN    NaN   NaN  NaN    NaN  
431836   NaN    NaN   NaN  NaN    NaN  
431837   NaN    NaN   NaN  NaN    NaN  

[431838 rows x 13 columns]

In [71]:
print(all_df.title.nunique())
print(all_df.title.unique())

1728
['0.0MHz' '1 더하기 3' '10 미니츠 곤' ... '프렌치 디스패치' '피치 오브 타임 극장판' '한창나이 선녀님']
